internal/process/manager.go

```diff
@@ -2,103 +2,115 @@ package process
 
+// ProcessManager is a way to manage processes across many different drivers / runtimes
+// Right now, all it does is hold a set of drivers and pass through to calls to the Driver interface for each of them
+// For that reason, we could consider removing it in the future and simply holding a map[node.Driver]Driver in the caller to this
 type ProcessManager interface {
-	ListProcesses() ([]*node.Process, error)
-	StartProcess(context.Context, *node.Process, *node.AppInstallation) error
+	// ListAllProcesses returns a list of all running process IDs, across all drivers
+	ListRunningProcesses(context.Context) ([]node.ProcessID, error)
+	// StartProcess starts a process for the given app installation with the given process ID
+	// It is expected that the driver can be derived from AppInstallation
+	StartProcess(context.Context, node.ProcessID, *node.AppInstallation) error
+	// StopProcess stops the process corresponding to the given process ID
 	StopProcess(context.Context, node.ProcessID) error
 	// Returns process state, true if exists, otherwise nil, false to indicate non-existence
-	GetProcess(node.ProcessID) (*node.Process, bool)
-	// ProcessManager should implement Component -- specifically, restore state given a set of processes and apps
+	IsRunning(context.Context, node.ProcessID) (bool, node.Driver, error)
+	// ProcessManager should implement Component -- specifically, restore state given by RestoreInfo
 	node.Component[RestoreInfo]
 }
 
+var (
+	ErrDriverNotFound = errors.New("no driver found")
+)
+
 type baseProcessManager struct {
-	processDrivers map[string]Driver
-	processes      map[node.ProcessID]*runningProcess
+	drivers map[node.Driver]Driver
 }
 
 func NewProcessManager(drivers []Driver) ProcessManager {
 	pm := &baseProcessManager{
-		processDrivers: make(map[string]Driver),
-		processes:      make(map[node.ProcessID]*runningProcess),
+		drivers: make(map[node.Driver]Driver),
 	}
 	for _, driver := range drivers {
-		pm.processDrivers[driver.Type()] = driver
+		pm.drivers[driver.Type()] = driver
 	}
 	return pm
 }
 
-func (pm *baseProcessManager) ListProcesses() ([]*node.Process, error) {
-	processList := make([]*node.Process, 0, len(pm.processes))
-	for _, process := range pm.processes {
-		processList = append(processList, process.Process)
+func (pm *baseProcessManager) ListRunningProcesses(ctx context.Context) ([]node.ProcessID, error) {
+	var allProcs []node.ProcessID
+	for _, driver := range pm.drivers {
+		procs, err := driver.ListRunningProcesses(ctx)
+		if err != nil {
+			return nil, err
+		}
+		allProcs = append(allProcs, procs...)
 	}
-	return processList, nil
+	return allProcs, nil
 }
 
-func (pm *baseProcessManager) GetProcess(processID node.ProcessID) (*node.Process, bool) {
-	proc, ok := pm.processes[processID]
-	if !ok {
-		return nil, false
+func (pm *baseProcessManager) IsRunning(ctx context.Context, id node.ProcessID) (bool, node.Driver, error) {
+	var derr error
+	for _, driver := range pm.drivers {
```

Comment:
We should have the process ID include a prefix for the driver type, so we don't have to loop through all the unrelated drivers each time. Especially since some of the drivers will call out to external processes.